README.md

```diff
@@ -1,69 +1,273 @@
+
+#### Deploying the Operator to a cluster
+
+The commands below will deploy the Custom Resource Definitions (CRDs), RBAC, the Operator, the replyURLSync and the example Ingress. The Operator should be fully operational after these commands have been executed.
+
+**Note:** The Makefile has the ability to build and push the container image manually, but there are GitHub Workflows in the `.github/workflows` folder that automate the process.
+
+1. Build and push the controller images to a container registry (this step can be skipped if you already have an image built and pushed):
+   
+   Replace `<some-registry>` with the container registry you would like to push the image to and `<tag>` with the tag to identify the image.
+
+   ```sh
+   make docker-build docker-push IMG=<some-registry>/reply-urls-operator:<tag>
+   ```
+
+2. Update the container image for the manager deployment:
+
+   Before deploying the Reply URLs Operator you will need to update the image being declared in the deployment file `config/manager/manager.yaml`. Update the `manager` container in the containers section of the file with the image you have built. 
+   
+   That section should look similar to the snippet below.
+
+   ```yaml
+         containers:
+         - name: manager
+           command:
+           - /manager
+           image: sdshmctspublic.azurecr.io/reply-urls-operator:prod-c4620b7-20220905093200
+           imagePullPolicy: Always
+   ```
+
+3. Create reply-urls-operator secret
+   Currently the App Registration's Client Secret for the Operator is set as an environment variable. You will need to create a secret called `reply-urls-operator` with a data object called `azure-client-secret`.
+
+   Command to create the secret manually:
+   ```sh
+    kubectl create secret -n admin generic reply-urls-operator --from-literal azure-client-secret=<client_secret>
+   ```
+
+4. Update the ReplyURLSync config:
+
+   There is a sample ReplyURLSync config in `config/samples/reply-url-sync-example.yaml` which can be update if needs be.
+
+   * ingressClassFilter: Name of the Ingress Class that you want to watch e.g. "traefik"
+   * domainFilter (optional): Regex of the domain of the Ingress Hosts you want to manage e.g. ".*.sandbox.platform.hmcts.net". Defaults to match all ".*"
+   * replyURLFilter (optional): Regex of the reply URLs you want to manage e.g. ".*.sandbox.platform.hmcts.net". This can be set to something different to the domainFilter if you would only like delete certain reply URLS from the app registration. Defaults to ".*"
+   * clientID: Client ID of the app registration you are authenticating with.
+   * objectID: Client ID of the app registration you want to sync ReplyURLS with.
+   * tenantID: Tenant ID of the app registration you are authenticating with.
```

Comment:
```suggestion
   * `ingressClassFilter`: Name of the Ingress Class that you want to watch e.g. "traefik"
   * `domainFilter` (optional): Regex of the domain of the Ingress Hosts you want to manage e.g. ".*.sandbox.platform.hmcts.net". Defaults to match all ".*"
   * `replyURLFilter` (optional): Regex of the reply URLs you want to manage e.g. ".*.sandbox.platform.hmcts.net". This can be set to something different to the domainFilter if you would only like to delete certain reply URLS from the app registration. Defaults to ".*"
   * `clientID`: Client ID of the app registration you are authenticating with.
   * `objectID`: Client ID of the app registration you want to sync ReplyURLs with.
   * `tenantID`: Tenant ID of the app registration you are authenticating with.
```